### PR TITLE
Update `get_gp_samples` to support input / outcome transforms

### LINKED ISSUE #1146: [Bug] Using GP samples on multioutput models with outcome transform produces strange samples
# 🐛 Bug
Hi there. I believe there's an issue with RFF GP samples (`get_gp_samples`) on a multioutput SingleTaskGP with outcome transforms.

## To reproduce
In a d=2, m=1 case everything is working as expected.
The transform is not considered in the GP samples, but can be applied manually giving a good agreement between ground truth, posterior mean and a GP sample drawn from the posterior.
m1

However, when adding another output (m=2)the GP samples are completely off from the posterior. (Solid lines show isocontours of the first output, while dashed lines show the second). The scale is completely off. As this does not happen without outcome transforms I'm suspecting this to be the cause.
m2

```python
# ground truth
def f1(x1, x2):
    return 2 * (x1 - 0.1) ** 2 + (x2 - 0.2) ** 2 + 1000

def f2(x1, x2):
    return (x1 - 0.9) ** 2 + 2 * (x2 - 0.7) ** 2 + 1000

X = torch.linspace(0, 1, 51)
X1, X2 = torch.meshgrid(X, X)
test_X = torch.stack([X1, X2], axis=-1).reshape(-1, 2)
train_X = torch.rand(20, 2)

# ----------------------------------------------------------------
# d=2, m=1 example
# ----------------------------------------------------------------
train_Y = f1(*train_X.T).reshape(-1, 1)

output_transform = Standardize(m=1)
model = SingleTaskGP(
    train_X,
    train_Y,
    outcome_transform=output_transform,
)
mll = ExactMarginalLogLikelihood(model.likelihood, model)
fit_gpytorch_model(mll)

# posterior mean
Ym = model.posterior(test_X).mean.reshape(51, 51, 1).detach()

# GP samples
gpsamples = get_gp_samples(model, num_outputs=1, n_samples=1)
Ys, _ = output_transform.untransform(gpsamples(test_X))
Ys = Ys.reshape(1, 51, 51, 1).detach()

fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(14, 6))
cs = ax1.contour(X1, X2, Ym[..., 0], levels=10)
ax1.clabel(cs, inline=True, fontsize=8)
ax1.set(xlabel="$x_1$", ylabel="$x_2$", title="Posterior mean")
cs = ax2.contour(X1, X2, Ys[0, ..., 0], levels=10)
ax2.clabel(cs, inline=True, fontsize=10)
ax2.set(xlabel="$x_1$", ylabel="$x_2$", title="GP sample")

# ----------------------------------------------------------------
# d=2, m=2 example
# ----------------------------------------------------------------
train_Y = torch.stack([f1(*train_X.T), f2(*train_X.T)], axis=-1)

output_transform = Standardize(m=2)
model = SingleTaskGP(
    train_X,
    train_Y,
    outcome_transform=output_transform,
)
mll = ExactMarginalLogLikelihood(model.likelihood, model)
fit_gpytorch_model(mll)

# posterior mean
Ym = model.posterior(test_X).mean.reshape(51, 51, 2).detach()

# GP samples
gpsamples = get_gp_samples(model, num_outputs=2, n_samples=1)
Ys, _ = output_transform.untransform(gpsamples(test_X))
Ys = Ys.reshape(1, 51, 51, 2).detach()

fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(14, 6))
cs = ax1.contour(X1, X2, Ym[..., 0], levels=10)
ax1.clabel(cs, inline=True, fontsize=8)
ax1.set(xlabel="$x_1$", ylabel="$x_2$", title="Posterior mean")
ax1.contour(X1, X2, Ym[..., 1], levels=10, linestyles="--")
cs = ax2.contour(X1, X2, Ys[0, ..., 0], levels=10)
ax2.contour(X1, X2, Ys[0, ..., 1], levels=10)
ax2.clabel(cs, inline=True, fontsize=10)
ax2.set(xlabel="$x_1$", ylabel="$x_2$", title="GP sample")

## Expected Behavior

`get_gp_samples` should also for the m=2 case with transforms give samples that are consistent with the posterior. 

## System information
botorch 0.6.3.1
gpytorch 1.6.0
torch 1.10.2+cpu

### PR DESCRIPTION
**Pull request by saitcakmak**
Summary:
Adds support for input & outcome transforms to `get_gp_samples`. Fixes #1146.

In addition, this changes it to return a non-batched model if `n_samples=1`.

NOTE:If using input / outcome transforms, the gp samples must be accessed via the gp_sample.posterior(X) call. Otherwise, gp_sample(X) will produce bogus values that do not agree with the underlying model. It is also highly recommended to use outcome transforms to standardize the input data, since the gp samples do not work well when training outcomes are not zero-mean.

Differential Revision: D36076992

### CODE REVIEW

**Review by Balandat — APPROVED** [review]
thanks!
